src/main/java/com/minecolonies/blockout/loader/ComponentConstructionController.java

```diff
@@ -0,0 +1,99 @@
+package com.minecolonies.blockout.loader;
+
+import com.minecolonies.blockout.controls.Image;
+import com.minecolonies.blockout.controls.ItemIcon;
+import com.minecolonies.blockout.controls.Label;
+import com.minecolonies.blockout.controls.button.ButtonImage;
+import com.minecolonies.blockout.controls.button.ButtonVanilla;
+import com.minecolonies.blockout.controls.text.Text;
+import com.minecolonies.blockout.controls.text.TextFieldVanilla;
+import com.minecolonies.blockout.core.Pane;
+import com.minecolonies.blockout.views.*;
+import com.minecolonies.blockout.views.scrolling.ScrollingGroup;
+import com.minecolonies.blockout.views.scrolling.ScrollingList;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.lang.reflect.Constructor;
+import java.util.HashMap;
+import java.util.Map;
+
+public class ComponentConstructionController
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add a private constructor to hide the implicit public one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1118)